tests/phpunit/CRM/Civicase/Hook/SelectWhereClause/LimitCaseQueryToAccessibleCaseCategoriesTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+use CRM_Civicase_Test_Fabricator_Contact as ContactFabricator;
+use CRM_Civicase_Test_Fabricator_CaseType as CaseTypeFabricator;
+use CRM_Civicase_Test_Fabricator_Case as CaseFabricator;
+
+/**
+ * Tests for limiting case query to accessible case categories.
+ *
+ * @group headless
+ */
+class CRM_Civicase_Hook_SelectWhereClause_LimitCaseQueryToAccessibleCaseCategoriesTest extends BaseHeadlessTest {
```

Comment:
Since most of the API calls from the FE comes with the case_type.category parameter pre-filled. Can we have tests that reflect this also. E.g a call requesting for cases of category awards when the user does not have basic award information. Or a call requesting for the cases of category award when the user has both basic case and basic award information permission, will both cases be returned or just the one the user requested for?